GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -274,29 +274,36 @@ void StartAsyncDataLoad()
             {
                 var cancellationToken = _asyncLoadCancellation.Next();
 
-                ThreadHelper.JoinableTaskFactory.RunAsync(() => LoadLinksForRevisionAsync(_revision)).FileAndForget();
+                ThreadHelper.JoinableTaskFactory.RunAsync(async () => { await LoadLinksForRevisionAsync(_revision); }).FileAndForget();
 
-                // No branch/tag data for artificial commands
-
-                if (AppSettings.CommitInfoShowContainedInBranches)
+                ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
                 {
-                    ThreadHelper.JoinableTaskFactory.RunAsync(() => LoadBranchInfoAsync(_revision.ObjectId)).FileAndForget();
-                }
+                    if (_refsOrderDict == null)
+                    {
+                        await LoadSortedRefsAsync();
+                    }
 
-                if (AppSettings.ShowAnnotatedTagsMessages)
-                {
-                    ThreadHelper.JoinableTaskFactory.RunAsync(() => LoadAnnotatedTagInfoAsync(_revision.Refs)).FileAndForget();
-                }
+                    // No branch/tag data for artificial commands
+                    if (AppSettings.CommitInfoShowContainedInBranches)
+                    {
+                        await LoadBranchInfoAsync(_revision.ObjectId);
+                    }
 
-                if (AppSettings.CommitInfoShowContainedInTags)
-                {
-                    ThreadHelper.JoinableTaskFactory.RunAsync(() => LoadTagInfoAsync(_revision.ObjectId)).FileAndForget();
-                }
+                    if (AppSettings.ShowAnnotatedTagsMessages)
```

Comment:
cancellationToken.ThrowIfCancellationRequested();

Reply:
But then `cancellationToken.ThrowIfCancellationRequested();` should be as the first line in `LoadAnnotatedTagInfoAsync`.

Reply:
We have awaited `LoadBranchInfoAsync` and now we want to make sure no cancellation was requested. `LoadBranchInfoAsync` does not know there was an async call before, hence we should do the check here.

Reply:
updated